inloop/solutions/views.py

```diff
@@ -164,11 +164,20 @@ def get(self, request, slug):
         })
 
 
-class NewSolutionArchiveView(LoginRequiredMixin, View):
+class SolutionArchiveView(LoginRequiredMixin, View):
+    def get_solution_or_404(self, solution_id):
+        # only the author himself and staff members
+        # are allowed to access the associated archive
+        if self.request.user.is_staff:
+            return get_object_or_404(Solution, pk=solution_id)
+        return get_object_or_404(Solution, pk=solution_id, author=self.request.user)
```

Comment:
That should be just be a (utility) function. Skip the class boilerplate, this will also solve inheriting `View` twice. Next, `get_solution_or_404` is not a good name because it doesn't tell the reader that privileged/staff users "see more" than ordinary users.

Reply:
But I must admit it's not easy to find a good and short name. Please add a doc comment which explains that staff users have broader access :-)

Reply:
Thanks for your review. Can you propose a name for the function? Otherwise I will leave the name as-is for now. When excluded into a utility function, I'll document the expected behavior. What about `access_solution_or_404`?